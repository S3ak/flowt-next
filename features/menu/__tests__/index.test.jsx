import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

import NavigationMenu from "../index";

const mockItems = [
  {
    label: "Splash",
    path: "/",
    isVisible: false,
  },
  {
    label: "Home",
    path: "/home",
    isVisible: true,
  },
  {
    label: "Activity",
    path: "/activity",
    isVisible: true,
  },
  {
    label: "Sign Up",
    path: "/sign-up",
    isVisible: false,
  },
];

const mockToggleNav = jest.fn();

jest.mock("@/libs/nav/useNav", () => {
  return jest.fn(() => ({
    items: mockItems,
    toggleNav: mockToggleNav,
  }));
});

describe("Snapshot | NavigationMenu", () => {
  it("renders Navigation unchanged", () => {
    const { container } = render(<NavigationMenu />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Unit | NavigationMenu", () => {
  it("renders a list of filtered links", async () => {
    render(<NavigationMenu />);

    fireEvent.click(screen.getByTestId("nav-toggle"));

    expect(screen.getByText("Home")).toHaveTextContent("Home");
    expect(screen.queryByText("Sign Up")).toBeNull();
    expect(mockToggleNav).toHaveBeenCalledTimes(1);
  });
});
