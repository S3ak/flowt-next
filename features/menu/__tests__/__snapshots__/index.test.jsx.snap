// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot | NavigationMenu renders Navigation unchanged 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 a {
  min-height: 45px;
  min-width: 100px;
  background: var(--c-action);
  color: var(--c-text--inverse);
  padding: var(--size-xs);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0px solid transparent;
  border-radius: var(--radius-m);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: 0;
  cursor: pointer;
}

.c7 a:hover {
  background: var(--c-action--hover);
}

.c0 {
  width: 100%;
  height: 100%;
  margin: auto;
  margin-right: 0;
  margin-left: 0;
  -webkit-animation: fadeIn;
  animation: fadeIn;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: var(--c-bg--inverse);
  border-radius: 5px 0px 0px 5px;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: var(--size-xxs);
  list-style: none;
  padding: inherit;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c5 a {
  display: block;
  padding: var(--size-s);
  min-height: 45px;
  min-width: 45px;
  cursor: pointer;
  color: var(--c-text--inverse);
}

.c5:hover {
  color: var(--c-action--hover);
}

.c1 {
  width: 100%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  width: 100%;
  width: 100%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  width: 100%;
  width: 100%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 45px;
  min-height: 45px;
  cursor: pointer;
  color: var(--c-bg);
}

<div
  aria-label="Mobile Menu"
  class="c0"
  role="navigation"
>
  <div
    class="c1"
  >
    <div
      aria-pressed="true"
      class="c2"
      data-testid="nav-toggle"
      role="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        />
      </svg>
    </div>
  </div>
  <div
    class="c3"
  >
    <ul
      class="c4"
    >
      <li
        class="c5"
      >
        <a
          href="/home"
        >
          Home
        </a>
      </li>
      <li
        class="c5"
      >
        <a
          href="/activity"
        >
          Activity
        </a>
      </li>
    </ul>
  </div>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <a
        href="/sign-in"
      >
        Logout
      </a>
    </div>
  </div>
</div>
`;
